Source/Blazorise/Components/Table/TableRow.razor.cs

```diff
@@ -84,19 +103,159 @@ protected Task OnMouseOverHandler( MouseEventArgs eventArgs )
         return MouseOver.InvokeAsync( EventArgsMapper.ToMouseEventArgs( eventArgs ) );
     }
 
+    internal void AddTableHeaderCell( TableHeaderCell tableHeaderCell )
+    {
+        if ( ParentTable.FixedColumnsSync )
+        {
+            tableHeaderCells?.Add( tableHeaderCell );
+            RecalculateHashCode();
+        }
+        else
+        {
+            SetFixedCellPosition( tableHeaderCell.Width, tableHeaderCell.FixedPosition, tableHeaderCell.SetFixedPositionStartOffset, tableHeaderCell.IncreaseFixedPositionEndOffset );
+        }
+    }
+
+    internal void RemoveTableHeaderCell( TableHeaderCell tableHeaderCell )
+    {
+        tableHeaderCells?.Remove( tableHeaderCell );
+    }
+
     internal void AddTableRowHeader( TableRowHeader tableRowHeader )
     {
-        SetFixedCellPosition( tableRowHeader.Width, tableRowHeader.FixedPosition, tableRowHeader.SetFixedPositionStartOffset, tableRowHeader.IncreaseFixedPositionEndOffset );
+        if ( ParentTable.FixedColumnsSync )
+        {
+            tableRowHeaders?.Add( tableRowHeader );
+            RecalculateHashCode();
+        }
+        else
+        {
+            SetFixedCellPosition( tableRowHeader.Width, tableRowHeader.FixedPosition, tableRowHeader.SetFixedPositionStartOffset, tableRowHeader.IncreaseFixedPositionEndOffset );
+        }
     }
 
-    internal void AddTableHeaderCell( TableHeaderCell tableHeaderCell )
+    internal void RemoveTableRowHeader( TableRowHeader tableRowHeader )
     {
-        SetFixedCellPosition( tableHeaderCell.Width, tableHeaderCell.FixedPosition, tableHeaderCell.SetFixedPositionStartOffset, tableHeaderCell.IncreaseFixedPositionEndOffset );
+        tableRowHeaders?.Remove( tableRowHeader );
     }
 
     internal void AddTableRowCell( TableRowCell tableRowCell )
     {
-        SetFixedCellPosition( tableRowCell.Width, tableRowCell.FixedPosition, tableRowCell.SetFixedPositionStartOffset, tableRowCell.IncreaseFixedPositionEndOffset );
+        if ( ParentTable.FixedColumnsSync )
+        {
+            tableRowCells?.Add( tableRowCell );
+            RecalculateHashCode();
+        }
+        else
+        {
+            SetFixedCellPosition( tableRowCell.Width, tableRowCell.FixedPosition, tableRowCell.SetFixedPositionStartOffset, tableRowCell.IncreaseFixedPositionEndOffset );
+        }
+    }
+
+    internal void RemoveTableRowCell( TableRowCell tableRowCell )
+    {
+        tableRowCells?.Remove( tableRowCell );
+    }
+
+    private void RecalculateHashCode()
+    {
+        elementHashCode = tableHeaderCells.GetListHash() + tableRowCells.GetListHash() + tableRowHeaders.GetListHash();
```

Comment:
In theory the `elementHashCode` might have duplicate hash. I think in that case, Blazor might cause an exception on the `@key` duplicate value.

Reply:
Any suggestion?

Reply:
`Guid.NewGuid()`? 😅

Reply:
hmm, or maybe append the elementId in that case?

Reply:
Yeah, that might work.